pkg/controlplane/control/manager.go

```diff
@@ -81,7 +87,37 @@ type Manager struct {
 	lock            sync.Mutex
 	serviceToImport map[string]types.NamespacedName
 
-	logger *logrus.Entry
+	podLock sync.RWMutex
+	// endpoints stores the IPv4/v6 address of cl-dataplane endpoints per namespace
```

Comment:
I see, That's good then. I will remove the namespace from the map. Do we need the esname, For multiple cl-dataplane pods, we might have a single cl-dataplane endpointslice, or multiple?  Since I am going to track only cl-dataplane endpointslice, and its limited to a single namespace, maybe I should just use an endpoint list

Reply:
Why do assume a single endpointslice?
It's up to kubernetes to decide this.
In particular:
```
By default, the control plane creates and manages EndpointSlices to have no more than 100 endpoints each. You can configure this with the --max-endpoints-per-slice [kube-controller-manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/) flag, up to a maximum of 1000.
```

Reply:
Are we expecting more than  100-1000 cl-dataplane endpoints per namespace? I feel it seems to complicate the logic, but we can support that requirement. However, would also like to understand what is the usual configuration. 
@elevran What do your thoughts?

Reply:
Done, Maintaining a map with esname and combining it when updating endpoint slices.